cmd/kubeadm/app/util/runtime/runtime_test.go

```diff
@@ -462,77 +468,45 @@ func TestDetectCRISocketImpl(t *testing.T) {
 	}
 }
 
-func TestPullImagesInParallelImpl(t *testing.T) {
-	testError := errors.New("error")
-
-	tests := []struct {
-		name            string
-		images          []string
-		ifNotPresent    bool
-		imageExistsFunc func(string) (bool, error)
-		pullImageFunc   func(string) error
-		expectedErrors  int
+func TestPullImagesInParallel(t *testing.T) {
+	t.Parallel()
+	for _, tc := range []struct {
+		name         string
+		ifNotPresent bool
+		prepare      func(*runtimefakes.FakeImpl)
+		shouldError  bool
 	}{
 		{
-			name:         "all images exist, no errors",
-			images:       []string{"foo", "bar", "baz"},
-			ifNotPresent: true,
-			imageExistsFunc: func(string) (bool, error) {
-				return true, nil
-			},
-			pullImageFunc:  nil,
-			expectedErrors: 0,
-		},
-		{
-			name:         "cannot check if one image exists due to error",
-			images:       []string{"foo", "bar", "baz"},
-			ifNotPresent: true,
-			imageExistsFunc: func(image string) (bool, error) {
-				if image == "baz" {
-					return false, testError
-				}
-				return true, nil
-			},
-			pullImageFunc:  nil,
-			expectedErrors: 1,
+			name: "valid",
 		},
 		{
-			name:         "cannot pull two images",
-			images:       []string{"foo", "bar", "baz"},
+			name:         "valid: ifNotPresent is true",
 			ifNotPresent: true,
-			imageExistsFunc: func(string) (bool, error) {
-				return false, nil
-			},
-			pullImageFunc: func(image string) error {
-				if image == "foo" {
-					return nil
-				}
-				return testError
-			},
-			expectedErrors: 2,
 		},
 		{
-			name:         "pull all images",
-			images:       []string{"foo", "bar", "baz"},
-			ifNotPresent: true,
-			imageExistsFunc: func(string) (bool, error) {
-				return false, nil
+			name: "invalid: pull fails",
+			prepare: func(mock *runtimefakes.FakeImpl) {
+				mock.PullImageReturns("", errTest)
 			},
-			pullImageFunc: func(string) error {
-				return nil
-			},
-			expectedErrors: 0,
+			shouldError: true,
 		},
-	}
+	} {
+		prepare := tc.prepare
+		shouldError := tc.shouldError
+		ifNotPresent := tc.ifNotPresent
 
-	for _, tc := range tests {
 		t.Run(tc.name, func(t *testing.T) {
-			actual := pullImagesInParallelImpl(tc.images, tc.ifNotPresent,
-				tc.imageExistsFunc, tc.pullImageFunc)
-			if len(actual) != tc.expectedErrors {
-				t.Fatalf("expected non-nil errors: %v, got: %v, full list of errors: %v",
-					tc.expectedErrors, len(actual), actual)
+			t.Parallel()
```

Comment:
i think we should avoid running tests in parallel.
benefits are minimal and we don't need to test any parallelism of test cases.
the kubeadm unit test suite runs for a total of 20 seconds after recent optimizations.

Reply:
Removed the parallel runs as suggested.